drivers/flash/CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ zephyr_library_sources_ifdef(CONFIG_FLASH_STM32_QSPI flash_stm32_qspi.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G flash_mcux_flexspi_mx25um51345g.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_ESP32 flash_esp32.c)
+zephyr_library_sources_ifdef(CONFIG_MTD_PARTITIONING mtd_part.c)
```

Comment:
I start thinking that this is more generic kind of driver and should be moved out of flash directory. It will be layer which can be gate to any memory, not only flash.

Reply:
In linux I think the flash drivers are found under mtd instead of flash, under mtd there is a mtd_core that does something similar as what we need. Some of my inspiration comes from there. The mtd (devices) require a erase routine, on other devices this is not required.